libraries/ms-common/inc/soft_timer.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <stdint.h>
+
+#include "status.h"
+
+#define SOFT_TIMER_NUM 10
+
+typedef void (*soft_timer_callback)(uint8_t timer_id, void* context);
+
+// Initializes a set of software timers. Clock speed should be in MHz. Subsequent calls will do
+// nothing. The clock speed is that of the external PLL crystal.
+void soft_timer_init(uint8_t clock_speed);
```

Comment:
Also, is there a specific reason this specifically checks if it's been initialized? It's very useful to be able to call the init function to reset the module, especially for testing.